packages/toplevel-alias/contracts/TopLevelAliasRegistry.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+/**
+ * @title TopLevelAliasRegistry
+ * @dev A contract for managing a registry of aliases mapped to names.
+ * This contract allows the owner to set aliases for given names, ensuring aliases end with '.eth',
+ * have at least three characters before '.eth', do not exceed a maximum length, and are not empty.
+ */
+contract TopLevelAliasRegistry is Ownable {
+    // Public mapping of names to aliases
+    mapping(string => string) public aliases;
+
+    // Maximum allowed length for an alias
+    uint256 private constant MAX_ALIAS_LENGTH = 100;
+
+    // Event emitted when an alias is set
+    event AliasSet(string indexed _name, string _alias);
+
+    /**
+     * @dev Constructor that sets the specified address as the owner of the contract.
+     * @param initialOwner The address to be set as the initial owner of the contract.
+     */
+    constructor(address initialOwner) Ownable(initialOwner) {
+    }
+
+    /**
+     * @dev Sets an alias for a given name, ensuring it meets various criteria including ENS validity.
+     * Only the owner of the contract can call this function.
+     * Validates that the name is not empty, the alias length is within limits, and ends with '.eth'.
+     *
+     * @param _name The name to map the alias to. Must not be empty.
+     * @param _alias The alias to be set for the given name. Must meet the criteria.
+     */
+    function setAlias(string memory _name, string memory _alias) public onlyOwner {
+        require(bytes(_name).length > 0, "Name cannot be empty");
+        require(bytes(_alias).length >= 7 && bytes(_alias).length <= MAX_ALIAS_LENGTH, "Alias length is invalid");
+        require(_endsWith(_alias, ".eth"), "Alias must end with '.eth'");
```

Comment:
I wonder if it would be sufficient to store only the alias label without the eth postfix onchain since its a top level alias that is always settled at ens. 

To compute the name of a domain 
1. Name template is already given `alice.${TLD}.eth`
2. Get chain from injected provider
3. use chainId of provider to fetch alias from contract
4. supply alias to template -> `alice.gno.eth`

Reply:
We could simply check if only one label exists in the provided string via ENS stringUtils lib https://github.com/ensdomains/ens-contracts/blob/5ae0cd7500215fa0c6c8baa7491a4bda1b1634dd/contracts/wrapper/NameWrapper.sol#L394